docs/content/latest/api/ysql/syntax_resources/ysql_grammar.ebnf

```diff
@@ -339,6 +339,15 @@ timestamp ::= '<DateTime Literal>' ;
 connlimit ::= '<Integer Literal>';
 uid ::= '<Text Literal>';
 
+(* 'CREATE' 'RULE' *)
+create_rule = 'CREATE' [ 'OR' 'REPLACE' ] 'RULE' rule_name 'AS' 'ON' rule_event
+              'TO' table_name [ 'WHERE' condition ]
+              'DO' [ 'ALSO' | 'INSTEAD' ] ( 'NOTHING' | command { ';' command } ) ;
```

Comment:
You're missing parentheses here:
```suggestion
              'DO' [ 'ALSO' | 'INSTEAD' ] ( 'NOTHING' | command | '(' command { ';' command } ')'  ) ;
```

Reply:
I think parentheses are unnecessary here.  Adding them not only doesn't change the diagram but also doesn't change the generated grammar.

Reply:
You're right; resolved.